tests/e2e/tests/simple/simple1_test.go

```diff
@@ -69,52 +65,42 @@ func TestMain(m *testing.M) {
 	os.Exit(tc.RunTest(m))
 }
 
-// TODO: need an "is cluster ready" before running any tests
-
 func TestSimpleIngress(t *testing.T) {
-	// Tests the rewrite/dropping of the /fortio/ prefix as fortio only replies
-	// with "echo debug server ..." on the /debug uri.
+	// Tests that a simple ingress with rewrite/dropping of the /fortio/ prefix
+	// works, as fortio only replies with "echo debug server ..." on the /debug uri.
 	url := tc.Kube.IngressOrFail(t) + "/fortio/debug"
-	log.Infof("Fetching '%s'", url)
-
-	retry := util.Retrier{
-		BaseDelay:   time.Second,
-		MaxDelay:    time.Second,
-		Retries:     1000,
-		MaxDuration: 105 * time.Second,
-	}
 
-	client := &http.Client{Timeout: 5 * time.Second}
-	_, err := retry.Retry(context.Background(), func(ctx context.Context, i int) error {
-		resp, err := ctxhttp.Get(ctx, client, url)
-		if err != nil {
-			log.Warnf("Attempt %d : http.Get error %v", i, err)
-			return fmt.Errorf("attempt %d : http.Get error %v", i, err)
+	// This checks until all of those 3 things become ready:
+	// 1) ingress pod is up (that done by IngressOrFail above)
+	// 2) the destination pod (and its envoy) is up/reachable
+	// 3) the ingress and routerules are propagated and active on the ingress
+	log.Infof("Fetching '%s'", url)
+	needle := []byte("echo debug server up")
+
+	o := fhttp.NewHTTPOptions(url)
+	var (
+		code      int
+		data      []byte
+		dataDebug string
+	)
+	timeOut := time.Now().Add(timeToWaitForIngress)
```

Comment:
any reason not to use `util.GetIngress(ns)` ?

Reply:
because of comment 2 and 3 ? (also what is the difference between `tc.Kube.IngressOrFail(t)` and `util.GetIngress(ns)` ?

Reply:
edit tc.Kube.IngressOrFail(t) eventually calls GetIngress

Reply:
OK. I guess it could be a bad assumption that getting an IP would take longer than the other two items.